website/addons/evernote/model.py

```diff
@@ -0,0 +1,128 @@
+
+from website.addons.base import AddonOAuthUserSettingsBase, AddonOAuthNodeSettingsBase
+from website.addons.base import StorageAddonBase
+from website.addons.evernote import (settings, utils)
+from website.addons.evernote.serializer import EvernoteSerializer
+from website.oauth.models import (ExternalProvider, OAUTH1)
+
+from modularodm import fields
+
+import logging
+logger = logging.getLogger(__name__)
+
+class Evernote(ExternalProvider):
+    """
+    First cut at the Evernote provider
+
+    """
+    name = 'Evernote'
+    short_name = 'evernote'
+
+    # Evernote has a sandbox and a production service
+    # with different endpoints
+
+    client_id = settings.EVERNOTE_CLIENT_ID
+    client_secret = settings.EVERNOTE_CLIENT_SECRET
+    _sandbox = settings.EVERNOTE_SANDBOX
+
+    BASE_URL = "https://www.evernote.com" if not _sandbox  \
+           else "https://sandbox.evernote.com"
+
+    auth_url_base = '{}/OAuth.action'.format(BASE_URL)
+    request_token_url = '{}/oauth'.format(BASE_URL)
+    callback_url = '{}/oauth'.format(BASE_URL)
+
+    _oauth_version = OAUTH1
+
+    def handle_callback(self, response):
+        """View called when the Oauth flow is completed. Adds a new BoxUserSettings
+        record to the user and saves the user's access token and account info.
+        """
+
+        client = utils.get_evernote_client(token=response.get('oauth_token'))
+
+        userStore = client.get_user_store()
+        user = userStore.getUser()
+
+        return {
+            'provider_id': user.id,  # or user.username
+            'display_name': user.name,
+            'profile_url': ''
+        }
+
+class EvernoteUserSettings(AddonOAuthUserSettingsBase):
+    oauth_provider = Evernote
+    serializer = EvernoteSerializer
+
+
+class EvernoteNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+    oauth_provider = Evernote
+    serializer = EvernoteSerializer
+
+    folder_id = fields.StringField(default=None)
+    folder_name = fields.StringField()
+    folder_path = fields.StringField()
+
+    # to hold data about current Notebook
+    _folder_data = None
+
+    def set_user_auth(self, user_settings):
+        """Import a user's Evernote authentication and create a NodeLog.
+        :param EvernoteUserSettings user_settings: The user settings to link.
+        """
+        self.user_settings = user_settings
+        #nodelogger = BoxNodeLogger(node=self.owner, auth=Auth(user_settings.owner))
+        #nodelogger.log(action="node_authorized", save=True)
+
+    def set_folder(self, folder_id, auth):
+        self.folder_id = str(folder_id)
+        self._update_folder_data()
+        self.save()
+
+        if not self.complete:
```

Comment:
This conditional doesn't need to exist. It's copied from an addon that uses folder-level OAuth grants, which EverNote doesn't need. When you're using the `generic_views`, `set_auth` will be called to give a node-level grant.